server/index.js

```diff
@@ -13,24 +15,20 @@ var bodyParser = require('body-parser');
 app.use(bodyParser.json()); // support json encoded bodies
 app.use(bodyParser.urlencoded({ extended: true })); // support encoded bodies
 
-
-app.get('/api/listings/:listings_id/questions', function(req, res) {
-  var allUsers = QuestionSet.find({primaryRecord: req.params.listings_id}).exec((err, data) => {
-    // IT DOES NOT APPEAR I CAN SORT BY ANYTHING WITHIN THE QUESTIONS ARRAY ('questions.numContributions'), AS IT IS TOO COMPLEX OF AN OBJECT FOR MONGOOSE TO SORT
-    // sort by primary record?
-    if (err) {
-      console.log(err)
-    } else {
-      res.send(data)
-    }
+app.get('/api/listings/:listings_id/questions', (req, res) => {
+  const id = req.params.listings_id; 
+  var query = `select * from questions FULL JOIN answers on answers.questionId = questions.id where questions.tripId = ${id}`;
+  db.getAllQuestionsWithTripId(query, (result) => {
```

Comment:
like I said above, I'd recommend setting up as a traditional error first callback so you can handle any errors that occur